### PR TITLE
fix fpm xml view localisation

### PR DESCRIPTION
**Pull request by devinea**
https://github.com/SAP/open-ux-tools/issues/1433

### CODE REVIEW

File: packages/fe-fpm-writer/src/page/custom.ts [hunk 1]
```diff
@@ -91,6 +91,14 @@ export function generate(basePath: string, data: CustomPage, fs?: Editor): Edito
     if (!fs.exists(viewPath)) {
         fs.copyTpl(join(root, 'ext/View.xml'), viewPath, config);
     }
+    // i18n.properties
+    const i18nPath = join(basePath, 'webapp', 'i18n', 'i18n.properties');
+    const i18TemplatePath = join(root, 'i18n', 'i18n.properties');
+    if (fs.exists(i18nPath)) {
```
Comment (815are) [inline packages/fe-fpm-writer/src/page/custom.ts:97]: @devinea 

One concern about hardcoded path to `i18n` file,
there can be scenario where we have `manifest.json` like:
```
"sap.ui5": {
        "models": {
            "i18n": {
                "type": "sap.ui.model.resource.ResourceModel",
                "uri": "custom-i18n2/i18n.properties"
            },
```

Then if there is such manifest and I create custom page using fpm-writer - in result new `i18n` entry or new file is created within `i18n/i18n.properties`. Maybe we could try detect path based on `manifest.json`?
I think custom i18n paths are rare scenario, not fully sure how critical it is

**Review by 815are — COMMENTED** [review]
@devinea 

I tested with app by triggering creation of custom pages.
I noticed following issues/questions:
1. Each time same entry `title` is created in `i18n.properties`
image

I think we should not create duplicate entry in `i18n` file.

2. Also if I create to continues custom pages - then there will be rendered same title for both custom page. Is it expected?

3.  Please check comment about hardcoded path to `i18n` file, there can be cases where `i18n` has different path configured in `manifest.json`,ut not sure how critical is that scenario

File: packages/fe-fpm-writer/src/page/custom.ts (hunk 1 above)
Comment (devinea) [reply packages/fe-fpm-writer/src/page/custom.ts:101]: @815are Thanks. Addressed in https://github.com/SAP/open-ux-tools/pull/1475/commits/0991f7d897b654b19de30916f8c99395b7347cfa

**Comment by devinea** [comment]
> 1. Each time same entry `title` is created in `i18n.properties`

@815are Good point. Fixed in  https://github.com/SAP/open-ux-tools/pull/1475/commits/047e78f388e878913e4fedb82ce48b5c2ff901b9

**Review by IainSAP — APPROVED** [review]
Thx @devinea 

- Approving with the caveat that we should be using general API for updating i18n as there are cases that wont be covered.

-  Tests updates as expected

**Review by 815are — COMMENTED** [review]
@devinea 

New scenario while testing manually:
1. There is option to create custom page using existing view. In such case unnecessary/unused(or duplicate) i18n entry is created.
I think you can use test -> https://github.com/SAP/open-ux-tools/blob/fix_fpm_xml_view_localization/packages/fe-fpm-writer/test/unit/page/custom.test.ts#L135
I think we need create new i18n entry only when new view.xml is generated/created

**Comment by devinea** [comment]
> 1. There is option to create custom page using existing view. In such case unnecessary/unused(or duplicate) i18n entry is created.

Thanks @815are Fixed in https://github.com/SAP/open-ux-tools/pull/1475/commits/31eb253db1d253a16e152626144b9510747a75db to only write i18nProperties when writing the view also.

**Review by 815are — APPROVED** [review]
Checked `fe-fpm-writer` part:
1. manual test - reported issues are fixed;
2. Small concern about edge cases with `i18n` path(other variations on how i18n path resolved), but perhaps can be improved/reused after https://github.com/SAP/open-ux-tools/issues/1365

**Review by IainSAP — APPROVED** [review]
Re-approving as already reviewed

**Comment by devinea** [comment]
Thanks @815are @IainSAP . I've made a note in https://github.com/SAP/open-ux-tools/issues/1365 to update this code with the common i18n function when available in open-ux-tools.
